autogen/main/cluster.tf.tmpl

```diff
@@ -152,6 +152,10 @@ resource "google_container_cluster" "primary" {
     dns_cache_config {
       enabled = var.dns_cache
     }
+
+    gce_persistent_disk_csi_driver_config {
```

Comment:
We should make this a dynamic block, as I've observed consistent issues with the behavior of addon blocks in `disabled` states.

Reply:
Right, i based this on the initial DNS cache PR. Did you want me to fix that addon in this PR too?

Reply:
```tf
-  cluster_gce_pd_csi_config = var.gce_pd_csi_driver ? [{ enabled = true }] : []
+  cluster_gce_pd_csi_config = var.gce_pd_csi_driver ? [{ enabled = true }] : [{ enabled = false }]
```

I couldn't disable this addon with the initial dynamic block approach (based on istio addon), one difference with the older addons is that newer addon config blocks (dns/csi) use `enabled` rather then `disabled` states. When trying to plan it doesn't indicate a change needs to be made when trying to flip back to disabled

I was only able to disable when forcing a dynamic block by proving a explicit `[{ enabled = false }]`. So the dynamic block is always going to trigger now, do you still want to keep this approach then?

Reply:
Hmm, that's odd. It might indeed by driven by the defaults.

Can you try importing an existing cluster (with this disabled) using the static block approach? If import is successful, then I'm fine with using that approach.